pkg/virt-controller/network/hotplug.go

```diff
@@ -52,14 +59,23 @@ func CalculateInterfacesAndNetworksForMultusAnnotationUpdate(vmi *v1.VirtualMach
 	return ifacesToAnnotate, networksToAnnotate, isIfaceChangeRequired
 }
 
-func ApplyDynamicIfaceRequestOnVMI(vm *v1.VirtualMachine, vmi *v1.VirtualMachineInstance, hasOrdinalIfaces bool) *v1.VirtualMachineInstanceSpec {
+func ApplyDynamicIfaceRequestOnVMI(networkClient k8scnicncfiov1.K8sCniCncfIoV1Interface, vm *v1.VirtualMachine, vmi *v1.VirtualMachineInstance, hasOrdinalIfaces bool) (*v1.VirtualMachineInstanceSpec, error) {
 	vmiSpecCopy := vmi.Spec.DeepCopy()
 	vmiIndexedInterfaces := vmispec.IndexInterfaceSpecByName(vmiSpecCopy.Domain.Devices.Interfaces)
 	vmIndexedNetworks := vmispec.IndexNetworkSpecByName(vm.Spec.Template.Spec.Networks)
 	for _, vmIface := range vm.Spec.Template.Spec.Domain.Devices.Interfaces {
 		_, existsInVMISpec := vmiIndexedInterfaces[vmIface.Name]
 		shouldBeHotPlug := !existsInVMISpec && vmIface.State != v1.InterfaceStateAbsent && (vmIface.InterfaceBindingMethod.Bridge != nil || vmIface.InterfaceBindingMethod.SRIOV != nil)
 		shouldBeHotUnplug := !hasOrdinalIfaces && existsInVMISpec && vmIface.State == v1.InterfaceStateAbsent
+		if shouldBeHotPlug || shouldBeHotUnplug {
+			if persistentIP, err := isPersistentIPNetwork(networkClient, vm.Namespace, vmIndexedNetworks[vmIface.Name]); err != nil {
+				return nil, err
+			} else if persistentIP {
+				log.Log.Errorf("Skipping hot{un}plug of persistentIP interface %s, not supported yet", vmIface.Name)
+				continue
+			}
+		}
```

Comment:
can't we move this logic to the new pkgs and just invoke from here the right func / method ?

Reply:
We can't out of the box because it uses `GetNetworkAttachmentDefinitionByName` (on network package)
and it will cause cycling dependency unless `GetNetworkAttachmentDefinitionByName` is moved somewhere else

Reply:
what if you change the signature and give it the nads from the controller ? 

I mean something like:
```golang
if shouldBeHotPlug || shouldBeHotUnplug {
    nads, err := GetNetworkAttachmentDefinitionByName(networkClient, namespace, []v1.Network{network})
    if err != nil {
        return false, fmt.Errorf("failed retrieving network attachment definitions: %w", err)
    }
    
    if persistentIP, err := newipamlib.IsPersistentIPNetwork(nads[network.Name], vm.Namespace, vmIndexedNetworks[vmIface.Name]); err != nil {
        return nil, err
    } else if persistentIP {
        log.Log.Errorf("Skipping hot{un}plug of persistentIP interface %s, not supported yet", vmIface.Name)
        continue
    }
}
```

Maybe it's not worth it; please sync w/ the maintainer.

Reply:
this can work, i will do that so it will reduce the finger print

Reply:
it means to inline this logic before the calling, lets wait and see if the maintainer prefer something over the other, imho it doesn't worth it

```
	if !vmispec.IsSecondaryMultusNetwork(network) {
		return false, nil
	}

	nads, err := GetNetworkAttachmentDefinitionByName(networkClient, namespace, []v1.Network{network})
	if err != nil {
		return false, fmt.Errorf("failed retrieving network attachment definitions: %w", err)
	}
```